### PR TITLE
Implement editable flag.

### LINKED ISSUE #6: Feature idea - Keep auto-behavior but make editable?
Thanks for your work on this, its exactly what I've tried to wrangle together on my own but does it in a much better way.

One use case that comes up for me however is the following:
- I have a component which, by default, should use a child reference for a field automatically because it covers 90% of use cases.
- However, the remaining 10% of the time i need to override that reference manually.

Basically, the feature would be something like:

    [SerializeField, Child(Flag.Editable)] ...

What do you think?  I'm happy to take a stab at implementing it.

### PR DESCRIPTION
**Pull request by minism**
Closes #6

### CODE REVIEW

File: SceneRefAttributeValidator.cs [hunk 1]
```diff
@@ -176,6 +176,16 @@ List<ReflectionUtil.AttributedField<SceneRefAttribute>> requiredFields
 
         private static object UpdateRef(SceneRefAttribute attr, Component c, FieldInfo field, object existingValue)
         {
+            if (attr.HasFlags(Flag.Editable))
+            {
+                // We convert the reference to a UnityEngine.Object for null checks.
+                if (existingValue as UnityEngine.Object != null)
```
Comment (KyleBanks) [inline SceneRefAttributeValidator.cs:182]: Unfortunately it's not quite that easy as we also need to handle arrays and interfaces. It should be enough to check null for objects and interfaces, and for arrays we'd need to check null plus length > 0

Comment (minism) [reply SceneRefAttributeValidator.cs]: Updated and tested to handle arrays, PTAL.

Comment (minism) [reply SceneRefAttributeValidator.cs]: Looks like interfaces still needs more work.

Comment (minism) [reply SceneRefAttributeValidator.cs]: Ok, fixed and tested for interfaces and arrays both.

**Review by KyleBanks — APPROVED** [review]
Awesome, thanks so much for this! Looks great.

**Pull request merged**